### ISSUE TITLE
Weird behavior for installed fonts after Windows 11 22H2 update

### ISSUE THREAD

**Issue by mkvlrn** [matthewjberger/scoop-nerd-fonts#198]
**TL;DR:**

## Fonts installed (they were installed before the update) appear to be installed but do not work - installing them manually from the cache folder works.

Not really sure if this should be opened here or elsewhere, could be a manifest problem? Here's what happened:

I have 2 NF fonts installed from this bucket: `firacode-nf-mono` for vscode and `hack-nf-mono` for the terminal. They are used with the names "FiraCode NFM" and "Hack NFM", respectively, and have been working fine for months.

Day after my windows 11 22h2 update, when I opened windows-terminal I've received the warning that "Hack NFM" coulndn't be found and it defaulted to Consolas. Checked `scoop list`, fonts still there. Checked the "Fonts" control panel applet, fonts are there.

But I cannot find them in a list, for example, trying to change Notepad's font. They are not there.

So I uninstall and reinstall both, restart computer, same problem.

BUT - after uninstalling them again, I go to scoops cache folder, see the `.ttf` files, and try to install them manually - by double clicking and hitting "Install". It worked! They appear exactly the same as before in the "Fonts" applet, but now are also listed on the font change dropdown for apps and work correctly as before.

Repeated the process 3 times to check my own sanity, it happened as described again and again.

**Comment by gitolicious** [comment]
Had the same problem and just now understood that it broke with the 22H2 update. Thanks for the workaround which at least temporarily fixes the problem.

**Comment by changyuheng** [comment]
I believe the root cause is: https://github.com/microsoft/terminal/issues/3257

**Comment by mkvlrn** [comment]
> I believe the root cause is: [microsoft/terminal#3257](https://github.com/microsoft/terminal/issues/3257)

It's not **only** on windows-terminal, it's system wide. It also happens to vscode, notepad, any applications. It's like the font was never installed to begin with.

**Comment by atlasmir** [comment]
Same issue here. I just did a fresh re-install of Windows 11 22H2. I then installed JetBrainsMono NF font via `scoop`. This step puts the font files into `%LOCALAPPDATA%\Microsoft\Windows\Fonts`. 

![image](https://user-images.githubusercontent.com/8758566/195615668-2286c0e6-d198-41ad-a97c-65f0a58a9d68.png)

I can even successfully set the font in Windows Terminal. The weirdest part is that the setting won't survive a system reboot.

After reboot:
- I navigated into `%LOCALAPPDATA%\Microsoft\Windows\Fonts`, and the font files are still there;
- I can still select the fonts in Windows Terminal settings, and the font preview in the settings page works well: ![image](https://user-images.githubusercontent.com/8758566/195616670-d2a261fe-ed3f-4990-91d9-9f94219395b9.png)
- However, Windows Terminal complained that the font could not be found when I clicked 'Save'. 
![image](https://user-images.githubusercontent.com/8758566/195617600-2f6d98d7-9170-4fa7-9e85-731d7166e161.png)
- Other apps, e.g., VS Code, could not recognize the fonts either.

**Comment by FDUZS** [comment]
Same issue here.

And for me, it happens after the installation of
**2022-10 Cumulative Update for Windows 11 Version 22H2 for x64-based Systems (KB5018427)**
yesterday.

**Comment by rozbo** [comment]
me too......

**Comment by atlasmir** [comment]
> It seems like this issue only happens to nerd-fonts like JetBrainsMono-NF (`scoop install JetBrainsMono-NF`), while JetBrains-Mono (`scoop install JetBrains-Mono`) has no such thing. Can you guys reproduce this?

For me, none of the fonts (installed by `scoop` into `%LOCALAPPDATA%\Microsoft\Windows\Fonts`) survive a reboot, including `JetBrains-Mono`.

**Comment by Dabombber** [comment]
There's nothing wrong with the font installation, it's an issue with the Windows font cache. You can use the following command to get them working again.
```PowerShell
 Restart-Service -Name FontCache
```

**Comment by atlasmir** [comment]
> You can use the following command to get them working again.

In my case, restarting Windows Font Cache Service, unfortunately, didn't resolve the issue.

**Comment by atlasmir** [comment]
Seems like Windows has a feature that can block untrusted fonts from loading outside of the `%windir%/Fonts` directory. However, the last time I checked, this feature was `Not configured` on my system.

https://learn.microsoft.com/en-us/windows/security/threat-protection/block-untrusted-fonts-in-enterprise

**Comment by DavidZhang73** [comment]
TL;DR `sudo scoop install -g <font>` instead of `scoop install <font>`
I encountered this issue after upgraded to 22H2, `Restart-Service -Name FontCache` does not help, `untrusted fonts` was also not configured, but https://github.com/microsoft/terminal/issues/14231#issuecomment-1280827973 seems to be a workable solution.
BTW, following that comment, I found https://github.com/matthewjberger/scoop-nerd-fonts/pull/161, which introduces installing font in user scope instead of system level. I suspect it conflicts with 22H2 upgrade. However, I have another desktop PC, which also upgraded to 22H2 recently, the font works fine.

**Comment by gitolicious** [comment]
Great hint, thanks a lot for that! Solved the problem for me.

**Comment by jonz94** [comment]
Hi there,

Sorry for being late...
and BIG THANKS to everyone here for providing so much useful information!

Base on the information from DavidZhang73 and #180.
It seems that the best solution for now is using `sudo scoop install -g <font>` to install font for all users.

**Comment by hungpham3112** [comment]
According to [this](https://github.com/microsoft/terminal/issues/9375) issue, it seems we need to install Nerd Fonts for all users instead of current user.

**Comment by felipecrp** [comment]
https://github.com/matthewjberger/scoop-nerd-fonts/pull/200 will probably solve this issue.

**Comment by mkvlrn** [comment]
Alright, so it's definitely not an issue in this repo, but the discussion finally got to a solution. Thanks! Closing this.

**Comment by iquiw** [comment]
@jonz94 Awesome!!
The first solution fixed the problem for me!

just a typo.

```diff
- $accessControlList = Get-Acl $userFontInstallDir
+ $accessControlList = Get-Acl $userFontInstallFolder
```
